llvm/lib/Transforms/Vectorize/VPlanAnalysis.cpp

```diff
@@ -564,8 +564,9 @@ SmallVector<VPRegisterUsage, 8> llvm::calculateRegisterUsageForPlan(
         } else {
           // The output from scaled phis and scaled reductions actually has
           // fewer lanes than the VF.
-          unsigned ScaleFactor = getVFScaleFactor(R);
-          ElementCount VF = VFs[J].divideCoefficientBy(ScaleFactor);
+          ElementCount VF = VFs[J];
+          if (unsigned ScaleFactor = getVFScaleFactor(R); ScaleFactor > 1)
+            VF = VF.divideCoefficientBy(ScaleFactor);
```

Comment:
nit: This can be:
```suggestion
          if (unsigned ScaleFactor = getVFScaleFactor(R))
            VF = VF.divideCoefficientBy(ScaleFactor);
```
because scaling by 1 should result in the same VF.

Reply:
Done.